packages/components/nodes/retrievers/HydeRetriever/HydeRetriever.ts

```diff
@@ -0,0 +1,121 @@
+import { VectorStore } from 'langchain/vectorstores/base'
+import { INode, INodeData, INodeParams } from '../../../src/Interface'
+import { HydeRetriever, HydeRetrieverOptions, PromptKey } from 'langchain/retrievers/hyde'
+import { BaseLanguageModel } from 'langchain/base_language'
+import { PromptTemplate } from 'langchain/prompts'
+
+class HydeRetriever_Retrievers implements INode {
+    label: string
+    name: string
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'Hyde Retriever'
+        this.name = 'HydeRetriever'
+        this.type = 'HydeRetriever'
+        this.icon = 'hyderetriever.svg'
+        this.category = 'Retrievers'
+        this.description = 'Use HyDE retriever to retrieve from a vector store'
+        this.baseClasses = [this.type, 'BaseRetriever']
+        this.inputs = [
+            {
+                label: 'Language Model',
+                name: 'model',
+                type: 'BaseLanguageModel'
+            },
+            {
+                label: 'Vector Store',
+                name: 'vectorStore',
+                type: 'VectorStore'
+            },
+            {
+                label: 'Prompt Key',
+                name: 'promptKey',
+                type: 'options',
+                options: [
+                    {
+                        label: 'websearch',
+                        name: 'websearch'
+                    },
+                    {
+                        label: 'scifact',
+                        name: 'scifact'
+                    },
+                    {
+                        label: 'arguana',
+                        name: 'arguana'
+                    },
+                    {
+                        label: 'trec-covid',
+                        name: 'trec-covid'
+                    },
+                    {
+                        label: 'fiqa',
+                        name: 'fiqa'
+                    },
+                    {
+                        label: 'dbpedia-entity',
+                        name: 'dbpedia-entity'
+                    },
+                    {
+                        label: 'trec-news',
+                        name: 'trec-news'
+                    },
+                    {
+                        label: 'mr-tydi',
+                        name: 'mr-tydi'
+                    }
+                ],
+                default: 'websearch'
+            },
+            {
+                label: 'Custom Prompt',
```

Comment:
@Yongtae723 here I added custom prompt:
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/9696f20c-fe48-4629-baf5-56f549b1c77a)

If custom prompt is set, promptKey will be ignored

Reply:
I confirm your code and this is what I intended!

thanks a lot!